### PR TITLE
tests: migrate tests that are only executed on xenial to bionic

### PR DESCRIPTION
**Pull request by sergiocazzolato**
Start using bionic to run tests that were executed just on xenial.

The idea is to start moving out from xenial which is not longer
supported.

### CODE REVIEW

File: tests/nested/core/core-snap-refresh-on-core/task.yaml [hunk 1]
```diff
@@ -5,7 +5,7 @@ details: |
     revision to a new one. It expects to find a new snap revision in the
     channel pointed by the NESTED_CORE_REFRESH_CHANNEL env var.
 
-systems: [ubuntu-16.04-64]
+systems: [ubuntu-18.04-64]
```
Comment (anonymouse64) [inline tests/nested/core/core-snap-refresh-on-core/task.yaml:8]: this test is specific to the core snap, so it naturally will have different behavior on uc16 than uc18, so can we just run this one on both systems?

**Review by anonymouse64 — COMMENTED** [review]
One test I think should still be run on uc16, but otherwise lgtm

**Review by mardy — APPROVED** [review]
LGTM!

**Review by bboozzoo — APPROVED** [review]
LGTM, thank you

**Pull request merged**